specs/eventing/overview.md

```diff
@@ -1,97 +1,146 @@
-multiple _Subscriptions_.
-
-See [Kind: Channel](spec.md#kind-channel).
-
----
-
-_Navigation_:
-
-- [Motivation and goals](motivation.md)
-- **Resource type overview**
-- [Interface contracts](interfaces.md)
-- [Object model specification](spec.md)
+The Knative Eventing API provides primitives for two common event-processing
+patterns:
+
+- Topology-based event routing ([`messaging.knative.dev`](#messaging))
+
+  Events are routed based on _connections between objects_ (in particular,
+  events flow along a [channel](#channel) to all
+  [subscriptions](#subscription)). This model can be thought of as "event
+  plumbing" in that events are managed like flows of water through pipes.
+
+- Content-based event routing ([`eventing.knative.dev`](#eventing))
+
+  Events are selected for routing based on the event _attributes_ rather than
+  primarily by object connections (a [broker](#broker) provides a stream of
+  events which can be selected by a [trigger](#trigger)). This model is more
+  akin to picking parts off a conveyor belt, where each event is considered
+  separately for processing.
+
+Knative Eventing does not directly specify mechanisms for other event-processing
+models, including multi-stage workflows, correlated request-reply, and
+sequential (windowed) event processing; these models could be built using the
+primitives provided by Knative, or Knative could deliver events to an external
+system that implements these models.
+
+In addition to the primitives needed to express the above patterns, Knative
+Eventing defines two [_interface contracts_](#interface-contracts) to allow
+connecting multiple types of Kubernetes objects as event senders and recipients
+to the core primitives.
+
+![Overview of objects](images/eventing-overview.svg)
+
+<!-- Generated from the following UML using PlantUML:
+@startuml
+' Remove shadows
+skinparam shadowing false
+' Set ordering
+left to right direction
+
+usecase Addressable <<Knative Service>>
+usecase Addressable AS A2 <<Kubernetes Service>>
+
+package "eventing.knative.dev" {
+agent Broker
+agent "Trigger" as T1
+agent "Trigger" as T2
+}
+
+Broker <-- T1 : broker
+Broker <-- T2 : broker
+Addressable <-u- T1 : subscriber
+
+package "messaging.knative.dev" {
+agent Channel
+agent "Subscription" as S1
+agent "Subscription" as S2
+}
+
+Channel <-- S1 :channel
+Channel <-- S2 :channel
+Addressable <-u- S2 :subscriber
+A2 <-u- S2 : reply
+@enduml
+-->
+
+## Interface Contracts
+
+In addition to the concrete types described below in the `messaging.knative.dev`
+and `eventing.knative.dev` API groups, Knative Eventing supports referencing
+objects in other API groups as destinations for event delivery. This is done by
+defining partial schemas which the referenced resources must support. The
+following interface contracts define resource fragments and partial schemas
+(required fields on an arbitrary API object) which form a basis for Knative
+Eventing.
```

Comment:
I wonder if we should have a pointer to the control-plane doc if people are interested in seeing the more precise definition of these objects?